client/packages/programs/src/JsonForms/components/Search/Search.tsx

```diff
@@ -4,15 +4,54 @@ import { withJsonFormsControlProps } from '@jsonforms/react';
 import { z } from 'zod';
 import { useZodOptionsValidation } from '../../common/hooks/useZodOptionsValidation';
 import { SearchWithUserSource } from './SearchWithUserSource';
+import { SearchWithDocumentSource } from './SearchWithDocumentSource';
+import { QueryValues } from './useSearchQueries';
 
-const Options = z.object({
+const UserOptions = z.object({
   /**
    * Source of the search data -- user input or extract it from document
    */
-  source: z.enum(['user', 'document']),
+  source: z.literal('user'),
```

Comment:
```suggestion
  type: z.literal('user'),
```

Think `type` is a better discriminator. Its also use elsewhere

Reply:
I don't really agree. "Type" is vague and could mean anything (and doesn't makes sense with "user"), whereas "source" is what is actually is.

Plus in the ui_schema, we already have "type" to describe what type of element it is ("Control", "Search", etc.) It's confusing to have "type" as an option as well.

Reply:
I meant like "Search type", e.g. type = "Document search" and "User search"

but yeah source would work as well. Type is more common as a discriminator though I would say. 

Yes looks slightly odd :)
``` js
{
  type: "Search"
  options: {
     type: "Document"
  }
}
```

don't really mind